cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3419,6 +3420,21 @@ private boolean isPodUpToDate(StatefulSet sts, Pod pod) {
                     });
         }
 
+        Future<ReconciliationState> cruiseControlApiSecret() {
+            if (this.cruiseControl != null) {
+                Future<Secret> secretFuture = secretOperations.getAsync(namespace, CruiseControl.apiSecretName(name));
+                return secretFuture.compose(res -> {
+                    if (res == null) {
+                        return withVoid(secretOperations.reconcile(reconciliation, namespace, CruiseControl.apiSecretName(name),
+                                cruiseControl.generateApiSecret()));
+                    }
+                    return withVoid(Future.succeededFuture(this));
+                });
+
+            }
+            return withVoid(secretOperations.reconcile(reconciliation, namespace, cruiseControl.apiSecretName(name), null));
```

Comment:
Can you use if/else? Using if/else is better because it allows to detect mistakes such as missing return etc.